docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,264 @@
+
+The liveness property below is functionally identical to the **[CH/Seq/2/Termination]** property we defined earlier:
+
+- **[CH/Proto/5/Termination]** If the initiator and responder are correct, then the handshake eventually terminates (i.e., both processes allocate a new `ConnectionEnd` object).
+
+**Remarks**.
+
+- Uniqueness.
+To guarantee uniqueness, we can assume there is a predicate `unique(v)` that processes invoke to check the uniqueness of input value `v`; this predicate returns true if the input value `v` is unique and otherwise returns false.
+
+- State verification.
+Like uniqueness, processes verify each other's state by calling on a predicate `verify(s)` that returns true iff the input state `s` passes verification.
+
+- State consistency.
+This property is for ensuring that processes are not out of sync.
+This property complements state verification: while verification ensures authenticity on the state of another process (_who_ is the other process), consistency ensures freshness on the state of another process (_what_ is the other process).
+We assume there is a predicate `consistent(s)` that returns true iff the input state `s` is consistent.
+
+### 2.2 Protocol
+
+> The following section is work-in-progres..
+
+The CH protocol consists of four datagrams that the initiator and responder exchange.
+We enumerate the names of these datagrams in the following code snippet.
+
+```golang
+type ConnHandshakeInit struct {  }
+type ConnHandshakeTry struct {  }
+type ConnHandshakeAck struct {  }
+type ConnHandshakeConfirm struct {  }
+```
+
+3. A `StateProof` is a data type that abstracts over the details of a cryptographic proof that any process can create, and another process can verify.
+Intuitively, a proof is helpful for some process `x` to guarantee to another process `y` that `x` is in a certain state.
+Typically, a `StateProof` object is valid with respect to a `State` object.
+
+The diagram below captures the handshake protocol at a high level, comprising the four datagram exchanges.
+Note the direction of each arrow, indicating which is the sender and the destination process for each datagram, respectively.
+
+```
+               The four messages comprising
+             IBC Connection Handshake Protocol
+
+     +-----------+   ConnHandshakeInit   +-----------+
+     |           |--------- 1 --------->>|           |
+     |           |                       |           |
+     |           |   ConnHandshakeTry    |           |
+     |           |<<------- 2 -----------|           |
+     | Initiator |                       | Responder |
+     |     I     |   ConnHandshakeAck    |     R     |
+     |           |--------- 3 --------->>|           |
+     |           |                       |           |
+     |           | ConnHandshakeConfirm  |           |
+     |           |<<------- 4 -----------|           |
+     +-----------+                       +-----------+
+```
+
+
+###### Protocol handlers
```

Comment:
It is probably useful taking a look at this high level spec: https://github.com/informalsystems/VDD/pull/5. We should probably try to align with protocol description style used there, i.e., defining process state and then set of functions (handlers) where preconditions and postconditions are defined. This avoids writing pseudocode (as TLA+ specs will have some) but put clear expectations.